frontend/src/components/CompareTable.tsx

```diff
@@ -49,27 +49,62 @@ const css = stylesheet({
   },
 });
 
+export interface xParentLabel {
+  label: string;
+  colSpan: number;
+}
+
 export interface CompareTableProps {
   rows: string[][];
   xLabels: string[];
   yLabels: string[];
+  xParentLabels?: xParentLabel[];
 }
 
 class CompareTable extends React.PureComponent<CompareTableProps> {
   public render(): JSX.Element | null {
-    const { rows, xLabels, yLabels } = this.props;
+    const { rows, xLabels, yLabels, xParentLabels } = this.props;
     if (rows.length !== yLabels.length) {
       logger.error(
         `Number of rows (${rows.length}) should match the number of Y labels (${yLabels.length}).`,
       );
     }
+
+    if (xParentLabels) {
+      const xParentLabelsLength = xParentLabels.reduce(
+        (length, xParentLabel) => length + xParentLabel.colSpan,
+        0,
+      );
+      if (xParentLabelsLength !== xLabels.length) {
+        logger.error(
+          `Number of columns with data (${xLabels.length}) should match the aggregated length of parent columns (${xParentLabelsLength}).`,
+        );
+      }
+    }
+
     if (!rows || rows.length === 0) {
       return null;
     }
 
     return (
       <table className={css.root}>
         <tbody>
+          {xParentLabels && (
```

Comment:
Should we prevent showing parent columns when `xParentLabelsLength` is not equal to `xLabels.length`?

Reply:
Sounds good - done! ✅ https://github.com/kubeflow/pipelines/pull/8024/commits/2d4e6057875c30c5d7e55bf06a563b37ff256733